libraries/stdlib/src/kotlin/concurrent/Timer.kt

```diff
@@ -4,66 +4,121 @@ import java.util.Timer
+ * (expressed in milliseconds) and with the interval of [period] milliseconds between the end of the previous task
+ * and the start of the next one.
+ *
+ * @param name the name to use for the thread which is running the timer.
+ * @param daemon if true, the thread is started as a daemon thread (the VM will exit when only daemon threads are running)
+ */
 public fun timer(name: String? = null, daemon: Boolean = false, initialDelay: Long = 0.toLong(), period: Long, action: TimerTask.() -> Unit): Timer {
     val timer = if (name == null) Timer(daemon) else Timer(name, daemon)
     timer.schedule(initialDelay, period, action)
     return timer
 }
 
+/**
+ * Creates a timer that executes the specified [action] periodically, starting at the specified [startAt] date
+ * and with the interval of [period] milliseconds between the end of the previous task and the start of the next one.
+ *
+ * @param name the name to use for the thread which is running the timer.
+ * @param daemon if true, the thread is started as a daemon thread (the VM will exit when only daemon threads are running)
+ */
 public fun timer(name: String? = null, daemon: Boolean = false, startAt: Date, period: Long, action: TimerTask.() -> Unit): Timer {
     val timer = if (name == null) Timer(daemon) else Timer(name, daemon)
     timer.schedule(startAt, period, action)
     return timer
 }
 
+/**
+ * Creates a timer that executes the specified [action] periodically, starting after the specified [initialDelay]
+ * (expressed in milliseconds) and with the interval of [period] milliseconds between the start of the previous task
+ * and the start of the next one.
+ *
+ * @param name the name to use for the thread which is running the timer.
+ * @param daemon if true, the thread is started as a daemon thread (the VM will exit when only daemon threads are running)
+ */
 public fun fixedRateTimer(name: String? = null, daemon: Boolean = false, initialDelay: Long = 0.toLong(), period: Long, action: TimerTask.() -> Unit): Timer {
     val timer = if (name == null) Timer(daemon) else Timer(name, daemon)
     timer.scheduleAtFixedRate(initialDelay, period, action)
     return timer
 }
 
+/**
+ * Creates a timer that executes the specified [action] periodically, starting at the specified [startAt] date
+ * and with the interval of [period] milliseconds between the start of the previous task and the start of the next one.
+ *
+ * @param name the name to use for the thread which is running the timer.
+ * @param daemon if true, the thread is started as a daemon thread (the VM will exit when only daemon threads are running)
+ */
 public fun fixedRateTimer(name: String? = null, daemon: Boolean = false, startAt: Date, period: Long, action: TimerTask.() -> Unit): Timer {
     val timer = if (name == null) Timer(daemon) else Timer(name, daemon)
     timer.scheduleAtFixedRate(startAt, period, action)
     return timer
 }
 
+deprecated("Use the built-in SAM conversion instead")
```

Comment:
Shouldn't we update usages in this file to invoke TimerTask constructor directly?